src/Microsoft.Extensions.Hosting/Internal/ServiceFactoryAdapter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting.Internal
+{
+    public abstract class ServiceFactoryAdapter
```

Comment:
Why an abstract class instead of an interface? There are no members/properties, default implementation and no non-public access modifiers.

Reply:
We've used abstracts elsewhere like security to have a base version of the same class without the generic. In this very simple scenario an interface could also be used. There's no strict reason either way.

Reply:
I think it's better to use an interface until there's proven need for an abstract base class. The base classes in Security seem to have properties or default method implementation which aren't possible on interfaces. The only reason to keep this as a class may be if we anticipate to add additional methods in the near future and I don't think there is reason to. Also even if it's trivial, let's keep the separate classes in separate files.

Reply:
Should this be true-internal? This isn't something anyone would actually be able to affect. We directly construct instances of it in the HostBuilder. The "abstraction" is not a public-facing abstraction, it's purely to handle a generic value.

Reply:
Agreed, make it internal.